framework/decode/vulkan_cpp_consumer_base.cpp

```diff
@@ -0,0 +1,2967 @@
+    uint32_t                                            format,
+    uint32_t                                            width,
+    uint32_t                                            height,
+    uint32_t                                            stride,
+    uint64_t                                            usage,
+    uint32_t                                            layers,
+    const std::vector<format::HardwareBufferPlaneInfo>& plane_info)
+{
+    if (platform_ == GfxToCppPlatform::PLATFORM_ANDROID)
+    {
+        FILE* file = GetFrameFile();
+
+        std::string android_hw_mem_name = "hHwBufferMemInfo" + std::to_string(memory_id);
+        AddKnownVariables("HardwareBufferMemoryInfo", android_hw_mem_name);
+        android_memory_id_map_[memory_id] = android_hw_mem_name;
+
+        VulkanCppAndroidBufferInfo buffer_info;
+        buffer_info.name      = "aHwBuffer" + std::to_string(buffer_id);
+        buffer_info.memory_id = memory_id;
+        AddKnownVariables("AHardwareBuffer*", buffer_info.name);
+        android_buffer_id_map_[buffer_id] = buffer_info;
+
+        fprintf(file, "\t{\n");
+        fprintf(file, "\t\tAHardwareBuffer_Desc desc = {};\n");
+        fprintf(file, "\t\tdesc.format               = %u;\n", format);
+        fprintf(file, "\t\tdesc.height               = %u;\n", height);
+        fprintf(file, "\t\tdesc.layers               = %u;\n", layers);
+        fprintf(file, "\t\tdesc.usage                = %" PRIu64 ";\n", usage);
+        fprintf(file, "\t\tdesc.width                = %u;\n", width);
+        fprintf(file, "\t\tAHardwareBuffer* buffer = nullptr;\n");
+        fprintf(file, "\t\tint              ahwbuf_res = AHardwareBuffer_allocate(&desc, &buffer);\n");
+        fprintf(file, "\t\tif ((ahwbuf_res == 0) && (buffer != nullptr))\n");
+        fprintf(file, "\t\t{\n");
+        fprintf(file, "\t\t\t%s = buffer;\n", buffer_info.name.c_str());
+        fprintf(file, "\t\t\tahwbuf_res = -1;\n");
+        fprintf(file, "\n");
+        fprintf(file, "\t\t\tstd::vector<format::HardwareBufferPlaneInfo> replay_plane_info;\n");
+        fprintf(file, "\n");
+        fprintf(
+            file,
+            "// The multi-plane functions are declared for API 26, but are only available to link with API 29.  So, "
+            "this\n");
+        fprintf(file, "// could be turned into a run-time check dependent on dlsym returning a valid pointer for\n");
+        fprintf(file, "// AHardwareBuffer_lockPlanes.\n");
+        fprintf(file, "#if __ANDROID_API__ >= 29\n");
+        fprintf(file, "\t\t\tif (desc.usage & AHARDWAREBUFFER_USAGE_CPU_WRITE_MASK)\n");
+        fprintf(file, "\t\t\t{\n");
+        fprintf(file, "\t\t\t\tAHardwareBuffer_Planes ahb_planes;\n");
+        fprintf(file, "\t\t\t\tahwbuf_res =\n");
+        fprintf(file,
+                "\t\t\t\t\tAHardwareBuffer_lockPlanes(buffer, AHARDWAREBUFFER_USAGE_CPU_WRITE_OFTEN, -1, "
+                "nullptr, &ahb_planes);\n");
```

Comment:
This code is not working.  I just encountered it being used and am working on fixing the issues with it.  So please currently ignore all "AHardwareBuffer" items in this file for now.

Reply:
Fixes have been applied.